vowpalwabbit/parse_args.cc

```diff
@@ -1593,7 +1601,9 @@ void merge_options_from_header_strings(
       }
     }
   }
-
+  if (is_ccb_run & !is_ccb_model) {
+    options.insert("ccb_model_without_has_seen_flag", "");
```

Comment:
This does not verify the has seen flag was not added though

Reply:
I think its only checking if you have the option ccb_explore_adf on the command line and ccb_explore_adf (or not) on the model file. So I'm assuming a model file that is CB related can be override to be ccb_explore_adf if you add that arg (and behind the scenes it adds ccb_model_without_has_seen_flag to make sure it can parse the model correctly). Is this interpretation correct Alexey?

Reply:
Kind of.
Before HasSeenMoreThanOneSlot flag was introduced  there was no difference between cb and ccb models. So basically cb models of vw from any version are equivalent (in terms of format) to ccb models of old vw. So logic which that we need here is the following:
if we are going to make ccb decisions, but input model is cb, it should be loaded as cb (~ legacy ccb) model.
In terms of options logic should be the following:
if model is cb and command line is ccb:
    set ccb_model_without_has_seen_flag

we can check if this flag is set already and not set it in this case, but not sure about benefit of that.